### DISCUSSION TITLE
[Bug] 使用 Docker 部署，使用域名访问报错

### DISCUSSION THREAD

**Discussion by macreee** [lobehub/lobehub#1460] [Self Hosting] [answered]
### 💻 系统环境 | Operating System

Windows

### 🌐 浏览器 | Browser

Chrome

### 🐛 问题描述 | Bug Description

**环境信息**
- Docker 部署
- Web 服务器 NGINX
- 域名访问
- CDN  cloudflare
**报错内容：**
```
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "default-src 'self' http: https: ws: wss: data: blob: 'unsafe-inline'".

    at Function (<anonymous>)
    at a (0fab221e.2e81d7bf95577bbb.js?__WB_REVISION__=2e81d7bf95577bbb:15:2714)
    at 0fab221e.2e81d7bf95577bbb.js?__WB_REVISION__=2e81d7bf95577bbb:15:3068
    at 0fab221e.2e81d7bf95577bbb.js?__WB_REVISION__=2e81d7bf95577bbb:15:3470
    at 0fab221e.2e81d7bf95577bbb.js?__WB_REVISION__=2e81d7bf95577bbb:1:745
    at 44022 (0fab221e.2e81d7bf95577bbb.js?__WB_REVISION__=2e81d7bf95577bbb:6251:23945)
    at i (webpack-550db4c573dd5429.js?__WB_REVISION__=MyKnlZIxmmAlPYNfzG2FK:1:167)
    at 11499 (1499.fb859b2f2e8fb4ea.js?__WB_REVISION__=fb859b2f2e8fb4ea:1:164)
    at Function.i (webpack-550db4c573dd5429.js?__WB_REVISION__=MyKnlZIxmmAlPYNfzG2FK:1:167)
```
使用域名访问会有上方报错，使用 IP+端口访问正常，想问下代码中不安全的部分未来是否有修改的可能？

### 🚦 期望结果 | Expected Behavior

访问正常

### 📷 复现步骤 | Recurrence Steps

使用问题描述中的环境信息复现即可

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
代码中应该没有不安全的脚本，有什么方案可以复现你的 case 的么

[a comment by an automated account is not shown]

**Comment by macreee** [comment]
> 代码中应该没有不安全的脚本，有什么方案可以复现你的 case 的么

忘记更新这个问题了，后来用 GPT 查了下是nginx 安全策略的问题，后来改了一下配置就可以了，具体修改内容晚点我贴上来。

[a comment by an automated account is not shown]

**Comment by heshunme** [comment, marked as answer]
> 代码中应该没有不安全的脚本，有什么方案可以复现你的 case 的么

当反向代理nginx包含了以下安全内容时可触发，发往服务端的聊天请求全部被浏览器本地拦截：
security.conf


\# security headers
add_header X-XSS-Protection          "1; mode=block" always;
add_header X-Content-Type-Options    "nosniff" always;
add_header Referrer-Policy           "no-referrer-when-downgrade" always;
add_header Content-Security-Policy   "default-src 'self' http: https: ws: wss: data: blob: 'unsafe-inline'; frame-ancestors 'self';" always;
add_header Permissions-Policy        "interest-cohort=()" always;
add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;

\# . files
location ~ /\.(?!well-known) {
    deny all;
}



另外，以上内容是nginxconfig.io网站默认生成的安全配置文件的内容。我在注释掉了该配置文件的导入后修复了问题。

[a comment by an automated account is not shown]

**Comment by heshunme** [comment]
> 代码中应该没有不安全的脚本，有什么方案可以复现你的 case 的么

我再次进行了测试，问题出在：
 add_header Content-Security-Policy "default-src 'self' http: https: ws: wss: data: blob: 'unsafe-inline'; frame-ancestors 'self';" always;

[a comment by an automated account is not shown]
